ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js

```diff
@@ -507,24 +508,51 @@ module.exports = function CesiumMap(
      */
     addRulerPoint(coordinates, markerLabel) {
       const { lat, lon } = coordinates
+
       // a point requires an altitude value so just use 0
       const point = [lon, lat, 0]
-      const options = {
+      const pointObject = convertPointCoordinate(point)
+      const cartographicPosition = Cesium.Cartographic.fromDegrees(
+        pointObject.longitude,
+        pointObject.latitude,
+        pointObject.altitude
+      )
+      let cartesianPosition = map.scene.globe.ellipsoid.cartographicToCartesian(
+        cartographicPosition
+      )
+      const billboardRef = billboardCollection.add({
+        image: DrawingUtility.getCircle({
+          fillColor: rulerColor,
+          icon: null,
+        }),
+        position: cartesianPosition,
         id: markerLabel,
-        title: `Selected ruler coordinate '${markerLabel}'`,
-        color: '#FC2803',
-        icon: null,
-        view: this,
+        eyeOffset,
+      })
+      //if there is a terrain provider and no altitude has been specified, sample it from the configured terrain provider
+      if (!pointObject.altitude && map.scene.terrainProvider) {
+        const promise = Cesium.sampleTerrain(map.scene.terrainProvider, 5, [
+          cartographicPosition,
+        ])
+        Cesium.when(promise, updatedCartographic => {
+          if (updatedCartographic[0].height && !this.isDestroyed) {
+            cartesianPosition = map.scene.globe.ellipsoid.cartographicToCartesian(
+              updatedCartographic[0]
+            )
+            billboardRef.position = cartesianPosition
+          }
+        })
       }
-      const useCustomText = true
+      map.scene.requestRender()
 
-      return this.addPointWithText(point, options, useCustomText)
+      return billboardRef
```

Comment:
❓ Can we still use the `addPointWithText()` method with a different set of options to avoid having duplicate code here?